experiments/GCM/heldsuarez.jl

```diff
@@ -0,0 +1,140 @@
+using LinearAlgebra
+using StaticArrays
+using Test
+using CLIMA
+using CLIMA.Atmos
+using CLIMA.LowStorageRungeKuttaMethod
+using CLIMA.Mesh.Filters
+using CLIMA.Mesh.Grids
+using CLIMA.MoistThermodynamics
+using CLIMA.PlanetParameters
+using CLIMA.VariableTemplates
+
+
+const ArrayType = CLIMA.get_array_type()
+
+Base.@kwdef struct HeldSuarezSetup{FT}
+    p_ground::FT = MSLP
+    T_initial::FT = 255
+    domain_height::FT = 30e3
+end
+
+function (setup::HeldSuarezSetup)(state, aux, coords, t) 
+    # callable to set initial conditions
+    FT = eltype(state)
+
+    r = norm(coords, 2)
+    h = r - FT(planet_radius)
+
+    scale_height = R_d * setup.T_initial / grav
+    p            = setup.p_ground * exp(-h / scale_height)
+
+    state.ρ  = air_density(setup.T_initial, p)
+    state.ρu = SVector{3, FT}(0, 0, 0)
+    state.ρe = state.ρ * (internal_energy(setup.T_initial) + aux.orientation.Φ)
+
+    return nothing
+end
+
+function config_heldsuarez(FT, N, resolution, setup)
+    config = CLIMA.GCM_Configuration("HeldSuarez", FT, ArrayType, setup,
+                                     ref_state=HydrostaticState(IsothermalProfile(setup.T_initial), FT(0)),
+                                     turbulence=ConstantViscosityWithDivergence(FT(0)),
+                                     moisture=DryModel(),
+                                     sources=(Gravity(),
+                                              Coriolis(),
+                                              held_suarez_forcing!))
+    CLIMA.GCM_SetupGrid!(config, N, resolution, setup.domain_height)
+
+    return config
+end
+
+function held_suarez_forcing!(source, state, aux, t::Real)
+    FT = eltype(state)
+
+    ρ     = state.ρ
+    ρu    = state.ρu
+    ρe    = state.ρe
+    coord = aux.coord
+    Φ     = aux.orientation.Φ
+    e     = ρe / ρ
+    u     = ρu / ρ
+    e_int = e - u' * u / 2 - Φ
+    T     = air_temperature(e_int)
+
+    # Held-Suarez constants
+    k_a       = FT(1 / (40 * day))
+    k_f       = FT(1 / day)
+    k_s       = FT(1 / (4 * day))
+    ΔT_y      = FT(60)
+    Δθ_z      = FT(10)
+    T_equator = FT(315)
+    T_min     = FT(200)
+
+    σ_b          = FT(7 / 10)
+    r            = norm(coord, 2)
+    @inbounds λ  = atan(coord[2], coord[1])
+    @inbounds φ  = asin(coord[3] / r)
+    h            = r - FT(planet_radius)
```

Comment:
Same as above, it would be good to provide an altitude function, e.g., in `orientation.jl`.